interface/main/calendar/add_edit_event.php

```diff
@@ -161,16 +161,34 @@ function DOBandEncounter()
 	 	    manage_tracker_status($event_date,$appttime,$temp_eid,$_POST['form_pid'],$_SESSION["authUser"],$_POST['form_apptstatus'],$_POST['form_room'],$encounter);
                  }
      }
-     else 
+     else
      {
-             # Capture the appt status and room number for patient tracker. 
+             # Capture the appt status and room number for patient tracker.
              if (!empty($_GET['eid'])) {
                 manage_tracker_status($event_date,$appttime,$_GET['eid'],$_POST['form_pid'],$_SESSION["authUser"],$_POST['form_apptstatus'],$_POST['form_room']);
              }
      }
     }
 
  }
+
+
+ /*This function is used for setting the date of the first event when using the "day_every_week" repetition mechanism.
+ When the 'start date' is not one of the days chosen for the repetition, the start date needs to be changed to the first
+ occurrence of one of these set days. */
+ function setEventDate($start_date, $recurrence){
+     $timestamp = strtotime($start_date);
+     $day = date('w', $timestamp);
+     //If the 'start date' is one of the set days
+     if(in_array(($day+1), explode(',',$recurrence))){
+         return $start_date;
+     }
+     //else: (we need to change start date to first occurrence of one of the set days)
+     $new_date = getNextEncounter($start_date, $recurrence);
```

Comment:
Done! Though I should mention that there are many references to "appointments" in this file which are written in the code as "encounters". That's actually where I got the idea of calling it 'encounters'...